### PR TITLE
cleanup on last merged PR + bugfix

### PR DESCRIPTION
**Pull request by digant73**
**Cleanup:**
* renamed all new levelcorner.bmp to level_corner.bmp
* renamed label levelcorner to level_corner
* renamed MKS_28_V1.0 to MKS_TFT28_V4.0 in platformio.ini
* added MKS_TFT28_V3.0 in platformio.ini
* updated config.ini with new MKS TFT32 V1.3 and MKS TFT28 V3.0
* usual code style cleanup on HAL folder
* some cleanup and optimization in the last merged PRs

**Improvement:**
* improved robustness of dialog box when used as splash screen

**PR STATUS:** ready to be merged

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -433,6 +434,27 @@ build_flags   = ${stm32f10x.build_flags}
   -DHSE_VALUE=25000000ul
   -DVECT_TAB_FLASH=0x08007000
   -DRAM_SIZE=64  ; Available RAM size in kbytes
-  -DHARDWARE="MKS_28_V1_0"
+  -DHARDWARE="MKS_28_V3_0"
   -DHARDWARE_SHORT="M28V10"
```
Comment (kisslorand) [inline platformio.ini:438]: M28V40

File: platformio.ini [hunk 2]
```diff
@@ -433,6 +434,27 @@ build_flags   = ${stm32f10x.build_flags}
...
+src_filter    = ${stm32f10x.default_src_filter} +<src/Libraries/Startup/stm32f10x_cl>
+extra_scripts = ${common.extra_scripts}
+                buildroot/scripts/stm32f107xC_0x7000_iap.py
+build_flags   = ${stm32f10x.build_flags}
+  -DSTM32F10X_CL=
+  -DHSE_VALUE=25000000ul
+  -DVECT_TAB_FLASH=0x08007000
+  -DRAM_SIZE=64  ; Available RAM size in kbytes
+  -DHARDWARE="MKS_28_V4_0"
+  -DHARDWARE_SHORT="M28V10"
```
Comment (kisslorand) [inline platformio.ini:459]: M28V40

File: platformio.ini (hunk 1 above)
Comment (digant73) [reply platformio.ini]: done

File: platformio.ini (hunk 2 above)
Comment (digant73) [reply platformio.ini]: done

**Pull request merged**